lib/types.nix

```diff
@@ -9,12 +9,13 @@ let
       name = "str";
       inherit description;
       descriptionClass = "noun";
-      check = lib.isString;
+      check = v: isString v || isRawType v;
       merge = lib.options.mergeEqualOption;
     };
+  isRawType = v: v ? __raw && isString v.__raw;
 in
 rec {
-  isRawType = v: lib.isAttrs v && lib.hasAttr "__raw" v && lib.isString v.__raw;
+  inherit isRawType;
```

Comment:
This wouldn't be needed if we built the `helpers` lib recursively, passing in `helpers` itself to `helpers`, like what nixpkgs/lib does.